src/query/models/tags.go

```diff
@@ -50,40 +49,23 @@ func EmptyTags() Tags {
 	return NewTags(0, nil)
 }
 
-// ID returns a string representation of the tags.
-func (t Tags) ID() string {
-	var (
-		idLen      = t.IDLen()
-		strBuilder = strings.Builder{}
-	)
-
-	strBuilder.Grow(idLen)
-	for _, tag := range t.Tags {
-		strBuilder.Write(tag.Name)
-		strBuilder.WriteByte(eq)
-		strBuilder.Write(tag.Value)
-		strBuilder.WriteByte(sep)
-	}
-
-	return strBuilder.String()
-}
-
-// IDMarshalTo writes out the ID representation
-// of the tags into the provided buffer.
-func (t Tags) IDMarshalTo(b []byte) []byte {
+// ID returns a byte slice representation of the tags.
+func (t Tags) ID() []byte {
```

Comment:
Can we also fix this ID so that it doesn't generate potentially colliding metric IDs?

i.e.
```
series 1, tags={"set_values":"something=foo,someother=bar"}
# generates this ID:
set_values=something=foo,someother=bar,

series 2, tags={"set_values":"something=foo","someother":"bar"}
# generates same ID:
set_values=something=foo,someother=bar,
```

In Prometheus they use `strconv.Quote(...)` to avoid this:
https://github.com/prometheus/tsdb/blob/master/labels/labels.go#L43-L58

You could refactor this method from the Go std library to use just []byte as well:
https://golang.org/src/strconv/quote.go

There's some sub-functions but it wouldn't be too difficult to do this properly by using a slightly modified version of it:
```go
func appendQuotedWith(buf []byte, s string, quote byte, ASCIIonly, graphicOnly bool) []byte {
	buf = append(buf, quote)
	for width := 0; len(s) > 0; s = s[width:] {
		r := rune(s[0])
		width = 1
		if r >= utf8.RuneSelf {
			r, width = utf8.DecodeRuneInString(s)
		}
		if width == 1 && r == utf8.RuneError {
			buf = append(buf, `\x`...)
			buf = append(buf, lowerhex[s[0]>>4])
			buf = append(buf, lowerhex[s[0]&0xF])
			continue
		}
		buf = appendEscapedRune(buf, r, quote, ASCIIonly, graphicOnly)
	}
	buf = append(buf, quote)
	return buf
}
```

Reply:
Good call. Would you recommend going this way or maybe doing something like pre-processing 
tag lengths and then appending it to the start of the ID, so it would look like: 

`series 1`
"set_values":"something=foo,someother=bar" -> `0.set_values:something=foo,someother=bar`

`series 2`
"set_values":"something=foo","someother=bar" -> `0,24.set_values:something=foosomeother:bar`